src/wallet/wallet.cpp

```diff
@@ -833,7 +835,7 @@ bool CWallet::GetAccountDestination(CTxDestination &dest, std::string strAccount
             bForceNew = true;
         else {
             // Check if the current key has been used
-            CScript scriptPubKey = GetScriptForDestination(account.vchPubKey.GetID());
+            CScript scriptPubKey = GetScriptForDestination(GetDestinationForKey(account.vchPubKey, g_address_type));
```

Comment:
If a key receives funds to an alternate address type, should the wallet force a new key?

Reply:
Perhaps, but is it worth implementing that for accounts (which are deprecated)?

Reply:
https://github.com/bitcoin/bitcoin/pull/11403#discussion_r159577303

> Perhaps, but is it worth implementing that for accounts (which are deprecated)?

This is also used by `getlabeladdress` in #7729. (If #11536 were merged, I think it would clear up some ongoing confusion about accounts and labels). But I think it still probably not worth changing the behavior in this PR, which is already doing enough things. Could have a comment here saying this code could check other address types besides `g_address_type` or store label/account addresses instead of pubkeys to be more reliable in the future.

Reply:
Added a TODO.